shared/utils/logic.ts

```diff
@@ -366,10 +263,101 @@ const isConditionFulfilled = (
  * @returns the condition field if it exists, `undefined` otherwise
  */
 const findConditionField = (
-  submission: LogicFieldSchemaOrResponse[],
-  fieldId: IConditionSchema['field'],
-): LogicFieldSchemaOrResponse | undefined => {
+  submission: LogicFieldOrResponse[],
+  fieldId: any,
```

Comment:
I don't think we should use `any` in types. Should this be `FormFieldDto['_id']` instead?